packages/lesswrong/components/community/modules/LocalGroups.tsx

```diff
@@ -0,0 +1,198 @@
+    lng: number,
+    known: boolean,
+    loading: boolean,
+  },
+  classes: ClassesType,
+}) => {
+  const { CommunityMapWrapper, CloudinaryImage2 } = Components
+  
+  /**
+   * Calculates the distance between the query location and the given lat/lng, as the crow flies
+   *
+   * @param {number} lat - latitude
+   * @param {number} lng - longitude
+   * @returns {number}
+   */
+  const distance = (lat, lng) => {
+    if (!userLocation) return null
+    
+    const toRad = (num) => num * Math.PI / 180
+    
+    const dLat = toRad(lat - userLocation.lat)
+    const dLng = toRad(lng - userLocation.lng)
+    const a = (Math.sin(dLat/2) * Math.sin(dLat/2)) + (Math.sin(dLng/2) * Math.sin(dLng/2) * Math.cos(toRad(userLocation.lat)) * Math.cos(toRad(lat)))
+    return Math.round(2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a)) * 6371)
+  }
+
+  let groupsListTerms: LocalgroupsViewTerms = {}
+  groupsListTerms = userLocation.known ? {
+    view: 'nearby',
+    lat: userLocation.lat,
+    lng: userLocation.lng,
+  } : {
+    view: 'local',
+  }
+  
+  const { results } = useMulti({
+    terms: groupsListTerms,
+    collectionName: "Localgroups",
+    fragmentName: 'localGroupsHomeFragment',
+    fetchPolicy: 'cache-and-network',
+    nextFetchPolicy: "cache-first",
+    limit: 200,
+    skip: userLocation.loading
+  });
+  
+  const cloudinaryCloudName = cloudinaryCloudNameSetting.get()
+
+  return (
+    <div className={classes.localGroups}>
+      <div className={classes.localGroupsList}>
+        {results?.map(group => {
+          const rowStyle = group.bannerImageId ? {
+            backgroundImage: `linear-gradient(to right, transparent, white 140px), url(https://res.cloudinary.com/${cloudinaryCloudName}/image/upload/c_fill,g_custom,h_115,w_140,q_auto,f_auto/${group.bannerImageId})`
+          } : {
+            backgroundImage: 'url(https://res.cloudinary.com/cea/image/upload/c_pad,h_80,w_140,q_auto,f_auto/ea-logo-square-1200x1200__1_.png), linear-gradient(to right, #e2f1f4, white 140px)'
+          }
+          
+          return <div key={group._id} className={classes.localGroup}>
+            <Link to={`/groups/${group._id}`} className={classes.mobileImg}>
+              {group.bannerImageId ?
+                <CloudinaryImage2 height={160} width="100vw" objectFit="cover" publicId={group.bannerImageId} imgProps={{w:'600'}} /> :
+                <img src="https://res.cloudinary.com/cea/image/upload/h_120,q_auto,f_auto/ea-logo-square-1200x1200__1_.png" />}
+            </Link>
+            <div className={classes.localGroupContent} style={rowStyle}>
+              <div className={classes.localGroupNameRow}>
+                <Link to={`/groups/${group._id}`} className={classes.localGroupName}>{group.name}</Link>
+                <div className={classes.localGroupDistance}>
+                  {userLocation.known && group.mongoLocation?.coordinates && `${distance(group.mongoLocation.coordinates[1], group.mongoLocation.coordinates[0])} km`}
```

Comment:
Hmm I realize I used km here, but maybe I should use miles by default, or let the user toggle between the two?

Reply:
Can't the browser do locale detection and present the right one? I bet there's an npm package for this.

Reply:
I added a toggle between km & mi. Defaults to km, unless your browser is set to `en-US`.

<img width="729" alt="Screen Shot 2022-02-24 at 1 53 57 PM" src="https://user-images.githubusercontent.com/9057804/155588928-dbddbbdd-03f5-4355-a67f-d18339958b74.png">

Reply:
<img src="https://media2.giphy.com/media/xT0xeuOy2Fcl9vDGiA/giphy-downsized-medium.gif"/>

Ok I'll include them too :)